libkbfs/config_local.go

```diff
@@ -1227,9 +1283,32 @@ func (c *ConfigLocal) SetTlfSyncState(tlfID tlf.ID, isSynced bool) error {
 	if isSynced {
 		diskCacheWrapped, ok := c.diskBlockCache.(*diskBlockCacheWrapped)
 		if !ok {
-			return errors.Errorf("Invalid disk cache type to set TLF sync state.")
+			return errors.New("invalid disk cache type to set TLF sync state")
+		}
+		if !diskCacheWrapped.IsSyncCacheEnabled() {
+			return errors.New("sync block cache is not enabled")
+		}
+	}
+	if !c.IsTestMode() {
```

Comment:
Might be nice to avoid indents below with:

```go
if c.IsTestMode() {
  return
}
```
instead.

Reply:
We still want to set the synced state in memory for test mode.